### PR TITLE
Import module

### PR DESCRIPTION
**Pull request by stevemessick**
Adds UI support for importing Flutter modules into Android apps.

Fixes build issues that arise with AS canary 7 (or maybe 5, not sure when it started).

Fixes plugin tool to work with AS canary 7 and Dart 2.0.

@devoncarew @pq @jwren

### CODE REVIEW

**Comment by devoncarew** [comment]
Looking now...

File: flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java [hunk 1]
```diff
@@ -52,17 +52,19 @@
...
-      if (System.getProperty("flutter.experimental.modules", null) != null) {
+      if (isNewProject) {
         res.add(new FlutterModuleGalleryEntry(sharedModel));
       }
     }
-    // TODO(messick): Create a new module type that will import an existing Flutter module into an Android app.
-    // It needs to edit /settings.gradle and /app/build.gradle as described in the add2app doc.
-    // The Project Structure editor will need to be extended to 'include' Flutter plugins for Flutter modules;
-    // without that users will have to edit /settings.gradle to import a Flutter plugin (which is true today).
+    else {
```
Comment (devoncarew) [inline flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java]: The else branch here looks like the one for `!projectHasFlutter`, but the comment below says ` isNewProject == false`

File: flutter-studio/src/io/flutter/module/FlutterModuleImporter.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    myRelativePath = io.flutter.utils.System.findRelativePath(projectRoot.getParent(), moduleRoot, File.separatorChar);
+    if (myRelativePath == null) {
+      showHowToEditDialog();
+      return;
+    }
+
+    myRelativePath = Paths.get(myRelativePath, ".android", "include_flutter.groovy").normalize().toString();
+    VirtualFile settingsFile = projectRoot.findChild("settings.gradle");
+    if (settingsFile == null) {
+      showHowToEditDialog();
```
Comment (devoncarew) [inline flutter-studio/src/io/flutter/module/FlutterModuleImporter.java]: 👍, I like that if things go off the rails, we give the user some manual step forward

File: flutter-studio/src/io/flutter/module/FlutterModuleImporter.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    writeLineToBuffer("    implementation project(':flutter')");
+    return true;
+  }
+
+  private void writeLineToBuffer(String line) {
+    writer.write(line);
+    writer.write(myNewline);
+  }
+
+  private static void showHowToEditDialog() {
```
Comment (devoncarew) [inline flutter-studio/src/io/flutter/module/FlutterModuleImporter.java]: Is this meant to be blank?

File: flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/fixture/IdeaFrameFixture.java [hunk 4]
```diff
@@ -492,15 +492,15 @@ private ActionButtonFixture findActionButtonByText(@NotNull String text) {
...
-  @NotNull
-  public AndroidToolWindowFixture getAndroidToolWindow() {
-    return new AndroidToolWindowFixture(getProject(), robot());
-  }
-
-  @NotNull
-  public CapturesToolWindowFixture getCapturesToolWindow() {
-    return new CapturesToolWindowFixture(getProject(), robot());
-  }
+  //@NotNull
```
Comment (devoncarew) [inline flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/fixture/IdeaFrameFixture.java]: Are these no longer needed? We should delete, or add a comment about why they're commented out for future readers.

File: flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java (hunk 1 above)
Comment (stevemessick) [reply flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java]: This is the block that adds items to the table shown when File>New>New Module is selected for an Android app project -- not Flutter. The only applicable item is "Import Flutter Module" and it isn't used anywhere else.

File: flutter-studio/src/io/flutter/module/FlutterModuleImporter.java (hunk 3 above)
Comment (stevemessick) [reply flutter-studio/src/io/flutter/module/FlutterModuleImporter.java]: No, it should show something to help the user make manual edits. I'm thinking I'll just keep it short and include a link to the add2app doc on github. That work for you? (I had been thinking of keeping it self-contained and including more detail but I think that would be rather annoying/distracting if a large dialog keeps popping up.)

Here's a possibility:
>The Gradle files could not be updated automatically. To use your Flutter module in this Android app you'll need to edit the files as described in the [add2app documentation](https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#make-the-host-app-depend-on-the-flutter-module).

File: flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/fixture/IdeaFrameFixture.java (hunk 4 above)
Comment (stevemessick) [reply flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/fixture/IdeaFrameFixture.java]: OK, I can delete them.

File: flutter-studio/src/io/flutter/module/FlutterModuleImporter.java (hunk 3 above)
Comment (devoncarew) [reply flutter-studio/src/io/flutter/module/FlutterModuleImporter.java]: 👍, sounds great re: keeping it simple and pointing people to the online docs

**Pull request merged**